File: Medium/Two Repeated Elements/two-repeated-elements.java
//{ Driver Code Starts
//Initial template for JAVA

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG
 {
	public static void main (String[] args) throws IOException
	 {
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	    int t = Integer.parseInt(br.readLine());
	    for(int i=0;i<t;i++){
	        int n = Integer.parseInt(br.readLine());
	        String l = br.readLine();
    		String[] sarr = l.split(" ");
    		int[] arr = new int[sarr.length];
    		for(int i1=0;i1<arr.length;i1++){
    			arr[i1] = Integer.parseInt(sarr[i1]);
    		}
    		
            Solution obj = new Solution();
            
            int[] res = obj.twoRepeated(arr, n);
            System.out.println(res[0] + " " + res[1]);
	    }
	}
}
// } Driver Code Ends


//User function template for JAVA

class Solution
{
    //Function to find two repeated elements.
    public int[] twoRepeated(int arr[], int n)
    {
        // Your code here
        Vector<Integer> v=new Vector<>();
        for(int i=0; i<n+2; i++){
            if(arr[Math.abs(arr[i])-1]<0){
                v.add(Math.abs(arr[i]));
            }
            else arr[Math.abs(arr[i])-1]=arr[Math.abs(arr[i])-1]*(-1);
        }
        int ans[]={v.get(0),v.get(1)};
        return ans;
    }
    
}

File: Medium/Two Repeated Elements/two-repeated-elements.py
#User function Template for python3

class Solution:
    
    #Function to find two repeated elements.
    def twoRepeated(self, arr , N):
        #Your code here
        res = []
        for elem in arr:
            abs_num = abs(elem)
            if arr[abs_num] < 0:
                res.append(abs_num)
            else:
                arr[abs_num] = -arr[abs_num]
        return res


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math

def main():
        T=int(input())
        while(T>0):
            
            N=int(input())

            A=[int(x) for x in input().strip().split()]
            
            obj = Solution()
            ans = obj.twoRepeated(A,N)
            print(ans[0], ans[1])
            
            T-=1


if __name__ == "__main__":
    main()
# } Driver Code Ends